# L2-Sea Propagation Benchmark Client

In this section we make model requests to the L2-Sea propagation benchmark. Make sure that the container is running in [here](https://github.com/MathSEE-Modeling-Week/Modeling-Week/blob/main/Models/L2-Sea/L2-benchmark_model.ipynb). We connect to the model and make a first model request. Remember that the Froude number (Fr) should be in the interval $[0.25, 0.41]$ and the draft (D) in the interval $[-6.776, -5.544]$. As a first client we choose the Froude number to be $Fr=0.32$ and draft $D=-6.2$.

In [9]:
import numpy as np
import umbridge

model = umbridge.HTTPModel("http://localhost:4242", "benchmark_UQ")# make sure that you connect to the "benchmark_UQ"

Fr = 0.32
D = -6.2

# Combine parameters into a single list
parameters = [Fr, D]

print(model([parameters]))

[[64.7470016]]


Thus, the result $R_T = 64.7470016$ represents the total resistance for given $Fr=0.32$ and $D=-6.2$. Next we make a MC simulation considering $N=100$ samples, which can take some time.

In [2]:
# Define the ranges for Froude number and draft
Fr_min, Fr_max = 0.25, 0.41
D_min, D_max = -6.776, -5.544

# Number of Monte Carlo samples
N = 100

# Generate random samples for Fr and D
Fr_samples = np.random.uniform(Fr_min, Fr_max, N)
D_samples = np.random.uniform(D_min, D_max, N)

# Store the results
results = []

for Fr, D in zip(Fr_samples, D_samples):
    parameters = [Fr, D]
    # Evaluate the model for each (Fr, D) pair
    output = model([parameters])
    # Safe output ,i.e. total resistance R_T
    results.append(output[0][0])
print

# Filter out the outlier
filtered_results = [x for x in results if x < 1e+3]

# Calculate the mean (Monte Carlo estimator)
mean_estimate = np.mean(filtered_results)

print(f"Monte Carlo Estimator (Mean) of the Total Resistance R_T: {mean_estimate}")

Monte Carlo Estimator (Mean) of the Total Resistance R_T: 73.157189086


Remember that the MC estimator is in this case given by

\begin{equation}
\hat{R_T} = \frac{1}{N} \sum_{i=1}^{N} R_T(Fr_i, D_i).
\end{equation}